workitem/workitem_repository.go

```diff
@@ -104,6 +104,10 @@ func (r *GormWorkItemRepository) Save(ctx context.Context, wi app.WorkItem) (*ap
 	if res.Version != wi.Version {
 		return nil, errors.NewVersionConflictError("version conflict")
 	}
+	// Changing the type of a work item is prohibited.
+	if res.Type != wi.Type {
+		return nil, errors.NewBadParameterError("type", wi.Type).Expected(res.Type)
```

Comment:
@aslakknutsen what do you mean? If you just pass the type that the WI already has, everything is fine. If you pass in some other type (which expresses an intent to modify it), the whole PATCH operation is cancelled just like as if the version doesn't match. Is there anything wrong with this approach? Reading up on JSON API PATCH requests now...